ape_hardhat/providers.py

```diff
@@ -195,11 +195,14 @@ def _make_request(self, rpc: str, args: list) -> Any:
     def set_block_gas_limit(self, gas_limit: int) -> bool:
         return self._make_request("evm_setBlockGasLimit", [hex(gas_limit)])
 
-    def sleep(self, seconds: int) -> int:
-        return int(self._make_request("evm_increaseTime", [seconds]))
-
-    def mine(self, timestamp: Optional[int] = None) -> str:
-        return self._make_request("evm_mine", [timestamp] if timestamp else [])
+    def set_timestamp(self, new_timestamp: int):
+        pending_timestamp = self.get_block("pending").timestamp
+        seconds_to_increase = new_timestamp - pending_timestamp
+        self._make_request("evm_increaseTime", [seconds_to_increase])
```

Comment:
I like the param to this might just be the timestamp and not seconds.. Let's try that and see

Reply:
I'll try this theory out